transformer_engine/pytorch/ops/sequential.py

```diff
@@ -129,11 +129,12 @@ def pop(self, idx: slice | int) -> torch.nn.Module:
         del self[idx]
         return out
 
-    def __iadd__(self, other: Sequential) -> Sequential:
-        return self.extend(other)
+    def __iadd__(self, modules: Iterable[torch.nn.Modules]) -> Sequential:
+        return self.extend(modules)
 
     def __add__(self, modules: Iterable[torch.nn.Modules]) -> Sequential:
-        out = self.__class__(self._modules)
+        out = Sequential()
+        out.extend(self)
         out.extend(modules)
         return out
```

Comment:
This implementation is similar to that in `torch.nn.Sequential`: https://github.com/pytorch/pytorch/blob/c64ad2403c0954a0c4a36c720f0c87b3d284d0c1/torch/nn/modules/container.py#L163-L175